patch-package
--- a/node_modules/azure-arm-apimanagement/lib/operations/user.js
+++ b/node_modules/azure-arm-apimanagement/lib/operations/user.js
@@ -741,6 +741,7 @@ function _createOrUpdate(resourceGroupName, serviceName, uid, parameters, option
     if (parameters !== null && parameters !== undefined) {
       let requestModelMapper = new client.models['UserCreateParameters']().mapper();
       requestModel = client.serialize(requestModelMapper, parameters, 'parameters');
+      requestModel.properties.identities = parameters.identities;
       requestContent = JSON.stringify(requestModel);
     }
   } catch (error) {
