patch-package
--- a/node_modules/azure-graph/lib/models/userCreateParameters.js
+++ b/node_modules/azure-graph/lib/models/userCreateParameters.js
@@ -57,6 +57,33 @@ class UserCreateParameters extends models['UserBase'] {
               name: 'String'
             }
           },
+          signInNames: {
+            required: false,
+            serializedName: 'signInNames',
+            type: {
+              name: 'Sequence',
+              element: {
+                required: false,
+                serializedName: 'SignInNameElementType',
+                type: {
+                  name: 'Composite',
+                  additionalProperties: {
+                    type: {
+                      name: 'Dictionary',
+                      value: {
+                        required: false,
+                        serializedName: 'ObjectElementType',
+                        type: {
+                          name: 'Object'
+                        }
+                      }
+                    }
+                  },
+                  className: 'SignInName'
+                }
+              }
+            }
+          },
           usageLocation: {
             required: false,
             serializedName: 'usageLocation',
@@ -99,6 +126,13 @@ class UserCreateParameters extends models['UserBase'] {
               name: 'String'
             }
           },
+          creationType: {
+            required: false,
+            serializedName: 'creationType',
+            type: {
+              name: 'String'
+            }
+          },
           passwordProfile: {
             required: true,
             serializedName: 'passwordProfile',
@@ -108,11 +142,11 @@ class UserCreateParameters extends models['UserBase'] {
                 type: {
                   name: 'Dictionary',
                   value: {
-                      required: false,
-                      serializedName: 'ObjectElementType',
-                      type: {
-                        name: 'Object'
-                      }
+                    required: false,
+                    serializedName: 'ObjectElementType',
+                    type: {
+                      name: 'Object'
+                    }
                   }
                 }
               },
@@ -120,7 +154,7 @@ class UserCreateParameters extends models['UserBase'] {
             }
           },
           userPrincipalName: {
-            required: true,
+            required: false,
             serializedName: 'userPrincipalName',
             type: {
               name: 'String'
